job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/step/StepContextSpec.groovy

```diff
@@ -3296,6 +3297,94 @@ class StepContextSpec extends Specification {
             skipTagLatest[0].value() == true
         }
         1 * jobManagement.requireMinimumPluginVersion('docker-build-publish', '1.0')
+        1 * jobManagement.logPluginDeprecationWarning('docker-build-publish', '1.2')
+    }
+
+    def 'call dockerBuildAndPublish with no options and version 1.2'() {
+        jobManagement.isMinimumPluginVersionInstalled('docker-build-publish', '1.2') >> true
+
+        when:
+        context.dockerBuildAndPublish {
+        }
+
+        then:
+        context.stepNodes.size() == 1
+        with(context.stepNodes[0]) {
+            name() == 'com.cloudbees.dockerpublish.DockerBuilder'
+            children().size() == 15
+            server[0].value().empty
+            registry[0].value().empty
+            repoName[0].value().empty
+            noCache[0].value() == false
+            forcePull[0].value() == true
+            dockerfilePath[0].value().empty
+            skipBuild[0].value() == false
+            skipDecorate[0].value() == false
+            repoTag[0].value().empty
+            skipPush[0].value() == false
+            createFingerprint[0].value() == true
+            skipTagLatest[0].value() == false
+            buildContext[0].value().empty
+            buildAdditionalArgs[0].value().empty
+            forceTag[0].value() == true
+        }
+        1 * jobManagement.requireMinimumPluginVersion('docker-build-publish', '1.0')
+        1 * jobManagement.logPluginDeprecationWarning('docker-build-publish', '1.2')
+    }
+
+    def 'call dockerBuildAndPublish with all options and version 1.2'() {
+        setup:
```

Comment:
Your code is not homogeneous between 2 tests: you use `setup` in this case but not in the other.
It wanted?

Use `given` instead of `setup`, seems to me is a better syntax, what do think ?